partitioner/src/main/java/io/confluent/connect/storage/partitioner/DefaultPartitioner.java

```diff
@@ -53,19 +58,21 @@ public String generatePartitionedPath(String topic, String encodedPartition) {
 
   @Override
   public List<T> partitionFields() {
+    if (partitionFields == null) {
+      partitionFields = newSchemaGenerator(config).newPartitionFields(PARTITION_FIELD);
+    }
     return partitionFields;
   }
 
   @SuppressWarnings("unchecked")
   public SchemaGenerator<T> newSchemaGenerator(Map<String, Object> config) {
     Class<? extends SchemaGenerator<T>> generatorClass = null;
     try {
-      generatorClass =
-          (Class<? extends SchemaGenerator<T>>) config.get(
-              PartitionerConfig.SCHEMA_GENERATOR_CLASS_CONFIG
-          );
+      generatorClass = (Class<? extends SchemaGenerator<T>>)
+          Class.forName(SCHEMA_GENERATOR_CLASS);
```

Comment:
Rather than use the class name constant here, what putting this line in a protected `Class<?
 extends SchemaGenerator<T> getSchemaGeneratorClass()` method? That way `TimeBasedPartitioner` ([here](https://github.com/confluentinc/kafka-connect-storage-common/pull/31/files#diff-35e4bfafabe9a14f5205e0b7bc968690R176)) could just override this method to return its preferred class rather than the more complex `newSchemaGenerator` method.

Not only does this allow `TimeBasedPartitioner` to inherit the more complex behavior, it also allows a custom subclass to do the same should they ever need it.